tiledb/sm/query/query.cc

```diff
@@ -1022,13 +1022,8 @@ Status Query::create_strategy() {
         array_schema_->allows_dups()) {
       use_default = false;
 
-      bool found = false;
-      bool non_overlapping_ranges = false;
-      RETURN_NOT_OK(config_.get<bool>(
-          "sm.query.sparse_unordered_with_dups.non_overlapping_ranges",
-          &non_overlapping_ranges,
-          &found));
-      assert(found);
+      auto&& [st, non_overlapping_ranges]{Query::non_overlapping_ranges()};
```

Comment:
The ranges aren't sorted until `dowork`. Do we need to move up the sorting to here?

Reply:
They get sorted in Subarray::non_overlapping_ranges